packages/dashboard/src/components/alert-store.tsx

```diff
@@ -1,77 +1,361 @@
+} from 'api-client';
 import { AppEvents } from './app-events';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  TextField,
+  Theme,
+  Divider,
+  useMediaQuery,
+  DialogContent,
+} from '@mui/material';
+import { makeStyles, createStyles } from '@mui/styles';
 import React from 'react';
+import { base } from 'react-components';
+import { AppControllerContext } from './app-contexts';
 import { RmfAppContext } from './rmf-app';
 import { Subscription } from 'rxjs';
-import { TaskAlertDialog } from './tasks/task-alert';
-
-// This needs to match the enums provided for the Alert model, as it is not
-// provided via the api-client since tortoise's pydantic_model_creator is used.
-enum AlertCategory {
-  Default = 'default',
-  Task = 'task',
-  Fleet = 'fleet',
-  Robot = 'robot',
+import { TaskCancelButton } from './tasks/task-cancellation';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    textField: {
+      background: theme.palette.background.default,
+      '&:hover': {
+        backgroundColor: theme.palette.background.default,
+      },
+    },
+  }),
+);
+
+interface AlertDialogProps {
+  alertRequest: AlertRequest;
+  onDismiss: () => void;
 }
 
-export const AlertStore = React.memo(() => {
+const AlertDialog = React.memo((props: AlertDialogProps) => {
+  const { alertRequest, onDismiss } = props;
+  const classes = useStyles();
+  const [isOpen, setIsOpen] = React.useState(true);
+  const { showAlert } = React.useContext(AppControllerContext);
   const rmf = React.useContext(RmfAppContext);
-  const [taskAlerts, setTaskAlerts] = React.useState<Record<string, Alert>>({});
-
-  const categorizeAndPushAlerts = (alert: Alert) => {
-    // We check if an existing alert has been acknowledged, and remove it from
-    // display.
-    if (alert.category === AlertCategory.Task) {
-      setTaskAlerts((prev) => {
-        const filteredTaskAlerts = Object.fromEntries(
-          Object.entries(prev).filter(([key]) => key !== alert.original_id),
-        );
+  const isScreenHeightLessThan800 = useMediaQuery('(max-height:800px)');
+  const [additionalAlertMessage, setAdditionalAlertMessage] = React.useState<string | null>(null);
+
+  const respondToAlert = async (alert_id: string, response: string) => {
+    if (!rmf) {
+      return;
+    }
+
+    try {
+      const resp = (
+        await rmf.alertsApi.respondToAlertAlertsRequestAlertIdRespondPost(alert_id, response)
+      ).data;
+      console.log(
+        `Alert [${alertRequest.id}]: responded with [${resp.response}] at ${resp.unix_millis_response_time}`,
+      );
+    } catch (e) {
+      const errorMessage = `Failed to respond [${response}] to alert ID [${alertRequest.id}], ${
+        (e as Error).message
+      }`;
+      console.error(errorMessage);
+      showAlert('error', errorMessage);
+      return;
+    }
+
+    const successMessage = `Responded [${response}] to alert ID [${alertRequest.id}]`;
+    console.log(successMessage);
```

Comment:
Add more context to the log, e.g. alert id.

Reply:
line 74 is where we construct the message, to be used in both a console log and a `showAlert`, it already contains the ID.